dashboard/urls.py

```diff
@@ -46,15 +49,22 @@
 
     # Thse URL's are data patterns
     # get file access patterns
-    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/grade_distribution', login_required(views.grade_distribution), name='grade_distribution'),
-    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/file_access_within_week', login_required(views.file_access_within_week), name='file_access_within_week'),
-    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/assignments', login_required(views.assignments), name='assignments'),
-    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/set_user_default_selection', login_required(views.update_user_default_selection_for_views), name='update_user_default_selection_for_views'),
-    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/get_user_default_selection', login_required(views.get_user_default_selection), name='get_user_default_selection'),
-    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/info', login_required(views.get_course_info), name='get_course_info'),
+    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/grade_distribution', 
+        cache_page(settings.CLIENT_CACHE_TIME)(login_required(views.grade_distribution)), name='grade_distribution'),
+    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/file_access_within_week', 
+        cache_page(settings.CLIENT_CACHE_TIME)(login_required(views.file_access_within_week)), name='file_access_within_week'),
+    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/assignments', 
+        cache_page(settings.CLIENT_CACHE_TIME)(login_required(views.assignments)), name='assignments'),
+    url(r'^api/v1/courses/(?P<course_id>[0-9]+)/set_user_default_selection', 
```

Comment:
This is actually a PUT call based on user default selection why this needs to be cached?

Reply:
Which URL are you referring to? I thought these were all GET's.

Reply:
`/set_user_default_selection`

Reply:
Oh you mean the set_user_default_selection? Yeah that probably should be excluded thanks for checking. I wasn't knew there might have been 1-2 that weren't GET's